### ISSUE TITLE
Map one single object

### ISSUE THREAD

**Issue by vryazanov** [dry-python/mappers#239]
Hey Artem,
nice to see you again.

Have a small question regarding a mapping for single object. Let's say I have a django project and have a `request.user` inside my view. So `User` instance is already here and I don't want to pass `user_id` to load function and call the db one more time. What's the easiest way to convert single instance of `User` to dataclass?

**Comment by proofit404** [comment]
Hi Vadim,

Ideally, you would not need to fetch a user instance at all. For example, https://github.com/dry-python/bookshelf/blob/master/bookshelf/middlewares.py

I understand other middlewares could require `request.user` to be set.

We could introduce `mapper.instantiate(UserModel())` method to the mapper.

**Comment by vryazanov** [comment]
Yep, `mapper.instantiate` sounds good to me. I can help and prepare a pull request if you have no objections

**Comment by proofit404** [comment]
Help would be highly appreciated! Thanks!

Our automatic release circle requires following a strict set of rules when we prepare a PR. We check it on CI using this file: [Dangerfile](https://github.com/dry-python/mappers/blob/master/dangerfile.ts) You can find all the rules if you search for `fail(` string. Shame on me for not documenting it yet.

We follow [semantic release](https://semantic-release.gitbook.io/semantic-release/) standard. You can run `npx commit` command (instead of `git commit`) to write commit messages properly. **Use :skip as the scope of the commit**. Usually, features have 3 commits `docs`, `test`, and `feat`.

The basic requirements are:
1. CI passed 
2. Danger passed
3. The feature is tested against all supported entity types and data sources.
4. A short page of the documentation is written

Also, I would like to discuss the necessary requirements for the test suite of this feature. In my opinion, it could contain something like:
* `instantiate` should return an instance of the entity type passed to the mapper
* `instantiate` should raise `InstanceError` if data source passed to the method has a different type from the data source passed to the mapper
* attributes of the entity should equal to attributes of the data source
* field rename should works (if the config of the field is str to str)
* related fields lookup should works (if the config of the field is str to a tuple)
* `instantiate` should raise `InstanceError` if data source passed to the method does not have attributes marked as evaluated in the mapper config
* `instantiate` could take `**kwargs`
  * if the data source does not have attribute marked as evaluated we could take the value from the `kwargs`
  * `kwargs` should take precedence on data source
  * user can't pass unknown field to the `kwargs`

Any thoughts?

Best regards,
Artem.

**Comment by vryazanov** [comment]
Thanks a lot for such detailed explanation, no thought from me for now :)
I'll dig deeper into the code and try to prepare draft pr by upcoming weekend.

**Comment by proofit404** [comment]
Cool! Thanks!
